### PR TITLE
[BUG-FIX] Fix Port Detection using VID/PID

### PR DESCRIPTION
**Pull request by haslinghuis**
This PR filters out VID/PID ports not associated with BF hardware. I have left the PortHandler detector intact as it works for 99% in other use cases.

Included an option for manufacturers and developers:
Screenshot from 2021-12-03 21-59-15

@mikeller requested to use id's instead of product names as names won't work in all instances.

@SteveCEvans This is a first attempt and needs testing as I don't have the hardware to confirm

More info: https://github.com/betaflight/betaflight/pull/10596

### CODE REVIEW

**Comment by SteveCEvans** [comment]
I can confirm that this works on MacOS with F4 (REVONANO) and H7 (MATEKH743) non-unified targets, and now only lists relevant devices.
image

Also tested on unified targets which worked before and are still working.
image

Thank you!

**Review by limonspb — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: src/js/serial.js
Comment (hydra) [inline src/js/serial.js:274]: `isFC` isn't true, not all USB to UART devices have an FC attached. Perhaps `isKnownSerialDevice` ?

**Comment by hydra** [comment]
I'm not sure this is a sustainable way to go or even a good idea.  It will prevent hardware manufacturers using other USB to UART adapters which are otherwise compatible and will hold manufacturers back to using old chips, which at some point will be NRND.

**Comment by haslinghuis** [comment]
@hydra We can add an option for manufacturers to show all ports. This PR only filters serial devices. DFU uses another depecreted API.
We could also add more VID/PID key values.

**Comment by haslinghuis** [comment]
@hydra DFU is already filtering ports display using VID/PID.

**Comment by hydra** [comment]
> @hydra We can add an option for manufacturers to show all ports. This PR only filters serial devices. DFU uses another depecreted API. We could also add more VID/PID key values.

I thought about that, but that's not really sustainable and would require additional communication between the manufacturer and all their users until any VID/PID combination was merged.

Given the recent release schedule of the configurator and BF itself, I would prefer solutions where users don't have to wait for maintainers to merge in code changes just to see a serial port, or to have to access some potential 'show all devices' setting.

DFU filtering seems less likely to change as the DFU VID/PID values come from MCU silicon and bootloader firmware, and the firmware only runs on a couple of variants.  When CPU or target support is added to the firmware the DFU change can be done at the same time, which makes sense release-schedule wise.

**Comment by haslinghuis** [comment]
Screenshot from 2021-12-03 21-59-15

**Comment by haslinghuis** [comment]
@hydra Added an option for manufacturers and developers. Still like to include new VIDs/PIDs in feature builds, but with this option nobody has to wait until it's in production.

**Comment by ctzsnooze** [comment]
It's not a bug fix.  This is a new feature imho.

**Comment by SteveCEvans** [comment]
@ctzsnooze This fixes the issue I addressed in https://github.com/betaflight/betaflight/pull/10596. Nobody else confirmed seeing the issue I saw but I found that non-unified targets would not auto connect using MacOS, possibly due to the other serial ports on that machine. This is a better solution which is why I closed my PR. I consider this a bug fix.

File: src/js/serial.js
Comment (haslinghuis) [reply src/js/serial.js]: Changed

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by ctzsnooze** [comment]
Works perfectly for me.
Shows only the ports I need, and that is nice!
The Manufacturer option that shows all available serial ports seems to work really well.
Seems to make the whole Configurator experience snappier on my machine.

**Review by ctzsnooze — APPROVED** [review]

**Review by limonspb — APPROVED** [review]

**Comment by hydra** [comment]
> @hydra Added an option for manufacturers and developers. Still like to include new VIDs/PIDs in feature builds, but with this option nobody has to wait until it's in production.

yeah, this is probably acceptable unless anyone can think of anything better.

**Review by chmelevskij — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> PASS
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by Asizon — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (PASS)
 - github identifies PR as mergeable -> PASS
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Pull request merged**
